### PR TITLE
Add baseFeePerGas to transaction state

### PR DESCRIPTION
**Pull request by vinistevam**
## Explanation
The extension `TransactionController` persists in a `baseFeePerGas` value for each confirmed transaction. Including this property into the transaction state in the core to support the unification initiative between the core and extension.

## References

Fixes https://github.com/MetaMask/MetaMask-planning/issues/1072

### `@metamask/transaction-controller`

- **ADDED**:  add `baseFeePerGas` to the transaction state and persists the value for each confirmed transaction

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/TransactionController.ts [hunk 1]
```diff
@@ -1169,8 +1169,12 @@ export class TransactionController extends BaseController<
           return [meta, false];
         }
 
+        const { gasUsed, baseFeePerGas } = txReceipt;
```
Comment (OGPoyraz) [inline packages/transaction-controller/src/TransactionController.ts:1172]: If I understood correctly, we should check if `txReceipt.blockNumber` exist, then another query to `getBlockByHash` to read that `baseFeePerGas` for that block.

Comment (vinistevam) [reply packages/transaction-controller/src/TransactionController.ts]: You are right, I thought I pushed this when I was cleaning up the commits. Anyway, it's pushed now, thanks!

**Review by OGPoyraz — APPROVED** [review]

File: packages/transaction-controller/src/TransactionController.ts [hunk 2]
```diff
@@ -1169,8 +1169,15 @@ export class TransactionController extends BaseController<
           return [meta, false];
         }
 
+        const txBlock = await query(this.ethQuery, 'getBlockByHash', [
+          txReceipt.blockHash,
+        ]);
+
         meta.verifiedOnBlockchain = true;
         meta.transaction.gasUsed = txReceipt.gasUsed;
+        if (txBlock && txBlock.baseFeePerGas) {
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:1178]: Is it worth simplifying to `meta.baseFeePerGas = txBlock?.baseFeePerGas;` ?

File: packages/transaction-controller/src/types.ts [hunk 3]
```diff
@@ -17,6 +17,7 @@ import type { Hex } from '@metamask/utils';
  * @property transaction - Underlying Transaction object
  * @property transactionHash - Hash of a successful transaction
  * @property blockNumber - Number of the block where the transaction has been included
+ * @property baseFeePerGas - The estimated block baseFeePerGas that will be burned. Introduced in EIP 1559. Value in hex wei
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/types.ts:20]:
```suggestion
 * @property baseFeePerGas - Base fee of the block as a hex value, introduced in EIP-1559.
```

Comment (matthewwalsh0) [inline packages/transaction-controller/src/types.ts:20]: It's hard to add detail to these going forward without `.` so I'm wondering if we should start using `.` going forward and add to the other properties for consistency?

File: packages/transaction-controller/src/types.ts [hunk 4]
```diff
@@ -43,6 +44,7 @@ type TransactionMetaBase = {
   blockNumber?: string;
   deviceConfirmedOn?: WalletDevice;
   verifiedOnBlockchain?: boolean;
+  baseFeePerGas?: Hex;
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/types.ts:47]: I've mentioned in the receipt PR that we can order these alphabetically so maybe a small conflict if that goes in first.

File: packages/transaction-controller/src/types.ts (hunk 3 above)
Comment (vinistevam) [reply packages/transaction-controller/src/types.ts]: I'm always unsure, but now that you raised it I vote for going forward using `.`.

**Review by matthewwalsh0 — APPROVED** [review]

**Review by OGPoyraz — DISMISSED** [review]

**Pull request merged**